Sources/SwiftProtobuf/Google_Protobuf_Timestamp+Extensions.swift

```diff
@@ -234,10 +234,26 @@ extension Google_Protobuf_Timestamp {
     ///
     /// - Parameter timeIntervalSince1970: The `TimeInterval`, interpreted as
     ///   seconds relative to 00:00:00 UTC on 1 January 1970.
+    @available(*, deprecated, renamed: "init(roundingTimeIntervalSince1970:rule:)")
     public init(timeIntervalSince1970: TimeInterval) {
-        let sd = floor(timeIntervalSince1970)
-        let nd = round((timeIntervalSince1970 - sd) * TimeInterval(nanosPerSecond))
-        let (s, n) = normalizeForTimestamp(seconds: Int64(sd), nanos: Int32(nd))
+        self.init(roundingTimeIntervalSince1970: timeIntervalSince1970, rule: .toNearestOrAwayFromZero)
+    }
+
+    /// Creates a new `Google_Protobuf_Timestamp` initialized relative to 00:00:00
+    /// UTC on 1 January 1970 by a given number of seconds, rounded to the nearest
+    /// nanosecond according to the given rounding rule.
+    ///
+    /// - Parameters:
+    ///   - imeIntervalSince1970: The `TimeInterval`, interpreted as
+    ///     seconds relative to 00:00:00 UTC on 1 January 1970.
+    ///   - rule: The rounding rule to use.
+    public init(
```

Comment:
@allevato and I were discussion options here, and this seems to follow things like `Date` in that it includes _time interval_ in the name.  Other things we considered:

```swift
init(
    roundingSince1970 timeInterval: TimeInterval,
    rule: FloatingPointRoundingRule
)
```

Which provides a slightly shorter name than what is suggested here.

```swift
init(
    timeIntervalSince1970: TimeInterval,
    roundingRule: FloatingPointRoundingRule
)
```

Would mean we don't have to deprecate the current method for the _default_ rounding, but the naming of the _rule_ argument is slightly different than foundation apis.

Reply:
I do quite like the new name, honestly.

Reply:
Ok, I guess we go with these.  I'll give the PR another day so folks approve or provide any other thoughts before merging it.